src/client/CodeEditor/widgets.ts

```diff
@@ -312,3 +313,73 @@ class ColorWidget extends WidgetType {
     return false;
   }
 }
+
+export const highlightWidgets = ViewPlugin.fromClass(
+  class {
+    showHighlight: boolean;
+    view: EditorView;
+    constructor(view: EditorView) {
+      this.showHighlight = false;
+      this.view = view;
+    }
+  },
+  {
+    decorations: (v) => {
+      if (!v.showHighlight) {
+        return Decoration.none;
+      }
+
+      const interactOpportunities = [];
+      const lines = v.view.state.doc.iter();
+      let line = lines.next();
+
+      //Offset is the number of characters before the regex so the highlighting can be placed properly.
+      let offset = 0;
+      while (!line.done) {
+        if (line.value === '\n') {
+          offset++;
+          line = lines.next();
+          continue;
+        }
+        const interactiveOccurances = line.value.matchAll(
+          //The below line contains all of the Regexes for all of the interactive widgets.
+          /\"\#([0-9]|[A-F]|[a-f]){6}\"|rotate\(-?\d*\.?\d*\)|https?:\/\/[^ "]+|vec2\(-?\b\d+\.?\d*\b\s*(,\s*-?\b\d+\.?\d*\b)?\)|true|false|(?<!\#)-?\b\d+\.?\d*\b|rgb\(.*\)/dg,
+        );
+        let interactOccurance =
+          interactiveOccurances.next();
+        while (!interactOccurance.done) {
+          const offsetInteract = interactOccurance.value;
+          offsetInteract.indices[0][0] += offset;
+          offsetInteract.indices[0][1] += offset;
+          interactOpportunities.push(offsetInteract);
+          interactOccurance = interactiveOccurances.next();
+        }
+        offset += line.value.length;
+        line = lines.next();
+      }
+      return Decoration.set(
+        interactOpportunities.map((opportunity) => {
+          return {
+            from: opportunity.indices[0][0],
+            to: opportunity.indices[0][1],
+            value: Decoration.mark({
+              class: 'cm-interact ',
+            }),
+          };
+        }),
+      );
+    },
+    eventHandlers: {
+      keydown(event, view) {
+        if (event.key == 'Alt') {
+          this.showHighlight = true;
```

Comment:
Aha! After setting this, it may be necessary to invoke some method to "update" the CodeMirror editor. I wonder if this might be the 2 second lag - maybe CodeMirror internally has some interval set where it checks for a "dirty" state and updates the display. Perhaps that could be triggered manually, but I'm not sure how.

Perhaps this one? https://codemirror.net/docs/ref/#view.WidgetType.updateDOM